integration-tests/chopsticks/src/tests/xcm.test.ts

```diff
@@ -0,0 +1,72 @@
+import { test, beforeAll, afterAll } from 'vitest'
+import { connectParachains, connectVertical, xcmLogger } from '@acala-network/chopsticks'
+import { sendTransaction, withExpect } from '@acala-network/chopsticks-testing'
+
+import * as SpiritnetNetwork from '../network/spiritnet.js'
+import * as PolkadotNetwork from '../network/polkadot.js'
+import * as HydraDxNetwork from '../network/hydroDx.js'
+import type { Config } from '../network/types.js'
+import { keysBob, keysCharlie } from '../helper.js'
+
+let spiritnetContext: Config
+let hydradxContext: Config
+let polkadotContext: Config
+
+beforeAll(async () => {
+	xcmLogger.level = 'info'
+	spiritnetContext = await SpiritnetNetwork.getContext()
+	hydradxContext = await HydraDxNetwork.getContext()
+	polkadotContext = await PolkadotNetwork.getContext()
+
+	await polkadotContext.dev.setStorage(PolkadotNetwork.defaultStorage(keysCharlie.address))
+	await spiritnetContext.dev.setStorage(SpiritnetNetwork.defaultStorage(keysBob.address))
+	await hydradxContext.dev.setStorage(HydraDxNetwork.defaultStorage(keysBob.address))
+
+	// Setup network
+	await connectVertical(polkadotContext.chain, spiritnetContext.chain)
+	await connectVertical(polkadotContext.chain, hydradxContext.chain)
+	await connectParachains([spiritnetContext.chain, hydradxContext.chain])
+
+	// fixes api runtime disconnect warning
+	await new Promise((r) => setTimeout(r, 500))
```

Comment:
Doesn't simply `await setTimeout(500)` work?

Reply:
`setTimeout` does not return a promise. Awaiting would be meaningless.

Reply:
yes looks much cleaner: [4f63eef](https://github.com/KILTprotocol/kilt-node/pull/614/commits/4f63eef43f9dc5410aa0090c0370bf81ea4229f4)